x-pack/platform/plugins/shared/security/server/authentication/api_keys/api_keys.ts

```diff
@@ -305,6 +324,85 @@ export class APIKeys implements APIKeysType {
     return result;
   }
 
+  private async uiamGrantApiKey(
+    authcScheme: string,
+    credential: string,
+    name: string,
+    expiration?: string
+  ) {
+    if (!this.uiam) {
+      throw new Error('UIAM service is not available');
+    }
+
+    this.logger.debug('Trying to grant an API key via UIAM');
+    this.logger.debug(`Using authorization scheme: ${authcScheme}`);
+
+    let result: GrantAPIKeyResult;
+
+    if (!credential.startsWith('essu_')) {
```

Comment:
Self Review: I know we discussed returning the ApiKey back if they tried to use a regular ApiKey for UIAM, but I just want to highlight for discussion